OpenRA.Mods.Common/Traits/SupportPowers/AirstrikePower.cs

```diff
@@ -48,6 +63,23 @@ public class AirstrikePowerInfo : SupportPowerInfo
 		[Desc("Weapon range offset to apply during the beacon clock calculation")]
 		public readonly WDist BeaconDistanceOffset = WDist.FromCells(6);
 
+		static object LoadSquad(MiniYaml yaml)
+		{
+			var ret = new List<AirstrikePowerSquadMember>();
+
+			// A less than great workaround for the YamlParser merging nodes and not supporting a way to remove nodes that are not "traits".
+			var squadNode = yaml.Nodes.FirstOrDefault(n => n.Key == "Squad@override") ?? yaml.Nodes.FirstOrDefault(n => n.Key == "Squad");
```

Comment:
This is really bad and hard to understand. I think the workaround should not live here, but rather in the rules without hardcoded MiniYaml keys.

```salt
^HQ:
	AirstrikePower:
		PauseOnCondition: lowpower
		Prerequisites: ~techlevel.superweapons
		Icon: airstrike
		ChargeInterval: 7500
		QuantizedFacings: 8
		Name: Air Strike
		Description: Deploy an aerial napalm strike.\nBurns buildings and infantry along a line.
		EndChargeSpeechNotification: AirstrikeReady
		SelectTargetSpeechNotification: SelectTarget
		InsufficientPowerSpeechNotification: InsufficientPower
		IncomingSpeechNotification: EnemyPlanesApproaching
		DisplayBeacon: True
		BeaconPoster: airstrike
		BeaconPosterPalette: beaconposter
		DisplayRadarPing: True
		CameraActor: camera
		ArrowSequence: arrow
		ClockSequence: clock
		CircleSequence: circles
		UseDirectionalTarget: True
		DirectionArrowAnimation: airstrikedirection
		SupportPowerPaletteOrder: 10
```

```salt
HQ:
	Inherits: ^HQ
	AirstrikePower:
		Squad:
			SquadMember@0:
				UnitType: a10
				SpawnOffset: -1536,-1536,0
				TargetOffset: 0,-1536,0
			SquadMember@1:
				UnitType: a10
			SquadMember@2:
				UnitType: a10
				SpawnOffset: -1536,1536,0
				TargetOffset: 0,1536,0
```

Reply:
I can agree, but that means messing with many missions' rules and even with the default skirmish rules.

Reply:
You could also make it 
```yaml
    AirstrikePower:
        UnitTypes: a10, a10, a10
        SpawnOffsets: -1536,-1536,0,  0,0,0,  0,-1536,0 
        TargetOffsets: 0,-1536,0,   0,0,0,    0,1536,0
```
I think literally everything is better than hardcoded node value names.

Reply:
I am putting the following excerpt from Discord here not to bitch and moan, but because I already said there that I'm not going to do that latest suggestion:
> I have been growing increasingly tired of, and am currently in no mood to keep doing things in a way I know is wrong suboptimal
Just because that's how we've been doing them for a decade
That doesn't mean I'm scrapping the PR entirely, but am currently not in the mood to throw out progress (even if it's for consistency with everything else)

That said, if all else fails I may be persuaded to do the first suggestion, with the inheritance restructuring.

As a primary candidate for a solution here, however, I would like to suggest what @abcdefg30 said on Discord (jokingly or not) - fixing the issue at the source - the YAML parser.
 - This line with the hacky node selection in the trait's Load function goes away
 - The mission rules change to:
```diff
airstrike.proxy:
	AirstrikePower:
-		Squad@override:
+		-Squad:
+		Squad:
			SquadMember:
				UnitType: a10
```
 - And we get the ability to remove YAML nodes on overrides.

If that's accepted, then I have a working prototype at [penev92/yamlOverrides](https://github.com/penev92/OpenRA/tree/yamlOverrides) (the third commit, really) and we can move this along.